data_structures/heap.cpp

```diff
@@ -0,0 +1,68 @@
+#include<iostream>
```

Comment:
Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..